### ISSUE TITLE
New project: lag when typing manually into project path input

### ISSUE THREAD

**Issue by snoopdouglas** [godotengine/godot#66620]
### Godot version

4.0-beta2

### System information

Fedora 36

### Issue description

![image](https://user-images.githubusercontent.com/4892574/193139334-0fa8089b-1761-4705-91d2-0f1922fc50b8.png)

In this state, whenever I backspace the 'a' - or type 'a' again after the / - Godot lags. I'm presuming this is because it's checking my home directory `/home/doug` for something.

Note that there are a lot of files in this directory - so maybe it's doing some sort of full scan?

### Steps to reproduce

Alternate the input between having an existent and non-existent path.

### Minimal reproduction project

n/a

**Comment by Calinou** [comment]
Can you reproduce this in 3.5?

**Comment by snoopdouglas** [comment]
A lot, over 10GB. This issue isn't present in 3.5.

**Comment by Nidjo123** [comment]
Did some testing with this, also using Fedora 36.

The lag starts happening  in Godot 4, alpha 11.

Bisect leads me to this commit: 53bd72fa705c79f568bd52d984f0a80eea7347a3. Seems to be related to window resizing, which happens on every `set_msg` call.

Specifically, in `DisplayServerX11::window_set_size` at the end there is a timeout loop:
```cpp
	XResizeWindow(x11_display, wd.x11_window, size.x, size.y);

	for (int timeout = 0; timeout < 50; ++timeout) {
		XSync(x11_display, False);
		XGetWindowAttributes(x11_display, wd.x11_window, &xwa);

		if (old_w != xwa.width || old_h != xwa.height) {
			break;
		}

		usleep(10000);
	}
```
which goes through all 50 timeouts for me and that causes the delay.

**Comment by Nidjo123** [comment]
Found out the reason for timing out. 

Noticed that it happens when trying to resize to a size that is outside of current min and max window sizes. Currently the size is set first, and then min and max so I moved the size setting at the end so it always falls into the given limits (we clamp between min and max so it is guaranteed).


Now there is another problem for which I can create a separate issue: the project dialog doesn't get resize events on X11 (haven't tested on other platforms) when resized programmatically (e.g. using `set_size`) which means the viewport doesn't get rendered properly, only the old size part of it does until you resize it manually or some other event triggers a window changed event. This also happens on current master branch without my changes.
